tests/modules/asset_groups/resources/test_create_asset_group.py

```diff
@@ -254,22 +277,52 @@ def test_create_asset_group_detection(
         data.add_filename(0, test_filename)
         data.set_field('speciesDetectionModel', ['someSortOfModel'])
 
-        resp_msg = 'failed to start detection'
-        asset_group_utils.create_asset_group(
-            flask_app_client, None, data.get(), 400, resp_msg
-        )
-        # TODO this is what the response should be
-        # resp_msg = 'Invalid submitter data'
-        # asset_group_utils.create_asset_group(
-        #    flask_app_client, None, data.get(), 403, resp_msg
-        # )
-
+        with mock.patch.object(
+            flask_app.acm,
+            'request_passthrough_result',
+            side_effect=detection_started,
```

Comment:
```diff
@@ -280,11 +280,25 @@ def test_create_asset_group_detection(
         with mock.patch.object(
             flask_app.acm,
             'request_passthrough_result',
-            side_effect=detection_started,
-        ):
+            return_value={'success': True},
+        ) as detection_started:
             resp = asset_group_utils.create_asset_group(
                 flask_app_client, None, data.get()
             )
+            assert detection_started.call_count == 1
+            assert detection_started.call_args[0][:-2] == (
+                'job.detect_request', 'post'
+            )
+            params = detection_started.call_args[0][-2]['params']
+            assert set(params.keys()) >= {
+                'endpoint',
+                'function',
+                'jobid',
+                'callback_url',
+                'image_uuid_list',
+                'input',
+            }
+            assert detection_started.call_args[0][-1] == 'cnn/lightnet'
             asset_group_uuid = resp.json['guid']
 
         from invoke import MockContext
```

It's better to do the asserts not in `side_effect` because you don't know whether the `side_effect` is actually called.

Reply:
Thanks, that's much cleaner. I was taking your example in the site_settings as a template.